pkg/service/aws/ecr.go

```diff
@@ -5,24 +5,49 @@ import (
 
 	"github.com/aws/aws-sdk-go/service/ecr"
 	"github.com/pkg/errors"
+	"gitlab.com/netbook-devs/spawner-service/pkg/service/labels"
 	proto "gitlab.com/netbook-devs/spawner-service/proto/netbookai/spawner"
 )
 
+func asEcrTags(label map[string]string) []*ecr.Tag {
+	tags := []*ecr.Tag{}
+	for k, v := range label {
+		key := k
+		val := v
```

Comment:
why not for `key, val := range label` ?

Reply:
we have to use the key, value as pointers, this is classic go gotchas. the `k` and `v` in loops are mutated and if we use pointer we end up using the wrong value.